src/components/form/SegmentRadioField/SegmentRadioField.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import { AppTheme } from '~/models/AppTheme ';
+import Colors from '~/theme/Colors';
+
+import {
+  StyledText,
+  StyledTouchableOpacity,
+  StyledView,
+} from './SegmentRadioField.styled';
+
+export interface SegmentRadioFieldProps {
+  touched?: boolean;
+  theme?: AppTheme;
+  value: any;
+  onPress(value: any): void;
+  setTouched(): void;
+  radioOptions: { label: string; value: React.ReactText }[];
+  color: string;
+}
+
+export const SegmentRadioField: React.FC<SegmentRadioFieldProps> = ({
+  onPress,
+  setTouched,
+  touched,
+  theme,
+  value,
+  radioOptions,
+  color,
+}) => {
+  return (
+    <StyledView>
+      {radioOptions.map((option, i) => {
+        const isSelected = value === option.value;
+        const activeBgColor =
+          theme === AppTheme.DARK ? Colors.navyBlue : '#ffffff';
+
+        return (
+          <StyledTouchableOpacity
+            key={i}
+            style={{
+              borderColor: isSelected ? color : Colors.silver,
+              backgroundColor: isSelected ? activeBgColor : 'transparent',
```

Comment:
If you would pass `isSelected` prop to `StyledTouchableOpacity` then you should be able to set those styled directly in the styled-component definition, am I right? :smile:

Reply:
Here's my solution for `StyledTouchableOpacity`:
```js
const StyledTouchableOpacity = styled.TouchableOpacity<{
  appTheme?: AppTheme;
  color: string;
  isSelected: boolean;
}>`
  background-color: ${({ appTheme, isSelected }) => {
    if (!isSelected) {
      return 'transpartent';
    }

    return appTheme === AppTheme.DARK ? Colors.navyBlue : '#ffffff';
  }};
  border-color: ${({ color, isSelected }) =>
    isSelected ? color : Colors.silver};
`;
```

Isn't it nice? :smile: